src/restore.c

```diff
@@ -131,39 +132,76 @@ do_restore_or_validate(time_t target_backup_id, pgRecoveryTarget *rt,
 	parray	   *parent_chain = NULL;
 	parray	   *dbOid_exclude_list = NULL;
 	bool        pgdata_is_empty = true;
-	bool        tblspaces_are_empty = true;
+	bool        cleanup_pgdata = false;
+	bool        backup_has_tblspc = true; /* backup contain tablespace */
 	XLogRecPtr  shift_lsn = InvalidXLogRecPtr;
 
+	if (instance_name == NULL)
+		elog(ERROR, "required parameter not specified: --instance");
+
 	if (params->is_restore)
 	{
 		if (instance_config.pgdata == NULL)
 			elog(ERROR,
 				"required parameter not specified: PGDATA (-D, --pgdata)");
+
 		/* Check if restore destination empty */
 		if (!dir_is_empty(instance_config.pgdata, FIO_DB_HOST))
 		{
+			/* if destination directory is empty, then incremental restore may be disabled */
```

Comment:
Do I understand it right, that incremental restore can be disabled solely for performance reasons?

Reply:
Yes, but not exactly. If there is no data to reuse, then there is no point in doing various checks, that we do to ensure, that incremental restore is safe to use. We can just roll it.